directives/tabs.py

```diff
@@ -0,0 +1,129 @@
+
+
+class ContentTabsDirective(Directive):
+    has_content = True
+    optional_arguments = 1
+
+    def run(self):
+        self.assert_has_content()
+        text = '\n'.join(self.content)
+        node = nodes.container(text)
+        node['classes'].append('content-tabs')
+
+        if self.arguments and self.arguments[0]:
+            node['classes'].append(self.arguments[0])
+
+        self.add_name(node)
+        self.state.nested_parse(self.content, self.content_offset, node)
+        return [node]
+
+
+class ContentTabsContainerDirective(Directive):
+    has_content = True
+    option_spec = {'title': directives.unchanged}
+    required_arguments = 1
+
+    def run(self):
+        self.assert_has_content()
+        text = '\n'.join(self.content)
+        node = nodes.container(text)
+        node['ids'].append('tab-%s' % self.arguments[0])
+        node['classes'].append('tab-content')
+
+        par = nodes.paragraph(text=self.options["title"])
+        par['classes'].append('tab-title')
+        node += par
+
+        self.add_name(node)
+        self.state.nested_parse(self.content, self.content_offset, node)
+
+        return [node]
+
+
+# Files added after triggering the 'builder-inited' event, in order to read the
+# settings from the conf.py file and prevent from reading the settings that
+# were added in the setup function
+def add_assets(app):
+    app.config.aplus_rst_tools_tabs_css and app.add_stylesheet(CSS_FILE)
+    app.config.aplus_rst_tools_tabs_js and app.add_javascript(JS_FILE)
+
+
+def copy_asset_files(app, exception):
+    if exception is None:
+        logger.info(
+            'Copying css files from the tabs directive to the _static folder... ')
+        # The files are added to the _build/html/_static/css folder.
+        if app.config.aplus_rst_tools_tabs_css:
+            html_static_path_css = os.path.join(assets_path, CSS_FILE)
+            local_path_css = resource_filename(__name__, html_static_path_css)
+            copy_asset(local_path_css, os.path.join(
+                app.outdir, assets_path, 'css'))
+
+        logger.info('done')
+
+        # The files are added to the _build/html/_static/js folder.
+        logger.info(
+            'Copying js files from the annotated directive to the _static folder... ')
+
+        if app.config.aplus_rst_tools_tabs_js:
+            html_static_path_js = os.path.join(assets_path, JS_FILE)
+            local_path_js = resource_filename(__name__, html_static_path_js)
+            copy_asset(local_path_js, os.path.join(
+                app.outdir, assets_path, 'js'))
+
+        logger.info('done')
+
+
+def setup(app):
+
+    app.add_config_value('aplus_rst_tools_tabs_css', True, 'html')
+    app.add_config_value('aplus_rst_tools_tabs_js', True, 'html')
+
+    app.add_directive('content-tabs',  ContentTabsDirective)
+    app.add_directive('tab-container', ContentTabsContainerDirective)
+
+    app.connect('builder-inited', add_assets)
```

Comment:
Using these signals is still a method for avoiding copying these files to the course build output directory if this tab directive is not used at all. Thus, I think we still want to use these signals instead of placing the JS/CSS files to the aplus theme static directory. That static directory would always be copied to the build output directory.